### PR TITLE
[cmake] Change set_swift_llvm_is_available to set definitions on a specific target.

### PR DESCRIPTION
**Pull request by gottesmm**
Otherwise we set it on all targets/languages in a subdirectory (I forgot if it
propagates up). Regardless, this type of viral stuff is something we want to
move away from since it creates a code that is a "forall" piece of code rather
than a piece of code that only effects a single target.

I also conditionalized the actual definitions being added on the compiled file's
language being C,CXX,OBJC,OBJCXX since as we add Swift sources to the host side
of the compiler, we will not want these flags to propagate to Swift sources.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -638,6 +638,7 @@ endmacro()
 
 # Declare that files in this library are built with LLVM's support
 # libraries available.
-macro(set_swift_llvm_is_available)
-  add_compile_options(-DSWIFT_LLVM_SUPPORT_IS_AVAILABLE)
+macro(set_swift_llvm_is_available name)
```
Comment (compnerd) [inline cmake/modules/AddSwift.cmake:641]: This should be changed to `function` from `macro` because it no longer pollutes the global namespace.

File: cmake/modules/AddSwift.cmake [hunk 2]
```diff
@@ -638,6 +638,7 @@ endmacro()
...
+  target_compile_options(${name} PRIVATE
+    $<$<COMPILE_LANGUAGE:C,CXX,OBJC,OBJCXX>:-DSWIFT_LLVM_SUPPORT_IS_AVAILABLE>)
```
Comment (compnerd) [inline cmake/modules/AddSwift.cmake:643]: This really should be `target_compile_definitions` and just drop the `-D` prefix.  However, this is a very odd thing to have to set on a per target basis.  Why is this needed?

Comment (gottesmm) [reply cmake/modules/AddSwift.cmake:643]: I don't know. I am just refactoring.

File: cmake/modules/AddSwift.cmake (hunk 1 above)
Comment (gottesmm) [reply cmake/modules/AddSwift.cmake:641]: agreed.

File: cmake/modules/AddSwift.cmake (hunk 2 above)
Comment (gottesmm) [reply cmake/modules/AddSwift.cmake:643]: I'll change it to target_compile_definition.

**Comment by gottesmm** [comment]
@swift-ci smoke test
